src/language-css/parser-postcss.js

```diff
@@ -92,29 +97,70 @@ function parseValueNodes(nodes) {
   return rootParenGroup;
 }
 
-function stringifyGroup(node) {
-  if (node.group) {
-    return stringifyGroup(node.group);
-  }
-
+function stringifyNode(node) {
   if (node.groups) {
-    return node.groups.reduce((previousValue, currentValue, index) => {
+    const open = (node.open && node.open.value) || "";
+    const groups = node.groups.reduce((previousValue, currentValue, index) => {
       return (
         previousValue +
-        stringifyGroup(currentValue) +
-        (currentValue.type === "comma_group" && index !== node.groups.length - 1
+        stringifyNode(currentValue) +
+        (node.groups[0].type === "comma_group" &&
+        index !== node.groups.length - 1
           ? ","
           : "")
       );
     }, "");
+    const close = (node.close && node.close.value) || "";
+
+    return open + groups + close;
   }
 
-  const before = node.raws && node.raws.before ? node.raws.before : "";
-  const value = node.value ? node.value : "";
-  const unit = node.unit ? node.unit : "";
-  const after = node.raws && node.raws.after ? node.raws.after : "";
+  const before = (node.raws && node.raws.before) || "";
+  const quote = (node.raws && node.raws.quote) || "";
+  const atword = node.type === "atword" ? "@" : "";
+  const value = node.value || "";
+  const unit = node.unit || "";
+  const group = node.group ? stringifyNode(node.group) : "";
+  const after = (node.raws && node.raws.after) || "";
+
+  return before + quote + atword + value + quote + unit + group + after;
+}
+
+function groupListContainsSassInterpolationMarker(groups) {
+  if (groups && groups.length) {
+    for (let i = groups.length - 1; i > 0; i--) {
+      // If we find `#{`, return true.
+      if (
+        groups[i].type === "word" &&
+        groups[i].value === "{" &&
+        groups[i - 1].type === "word" &&
+        groups[i - 1].value.endsWith("#")
+      ) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+function groupListContainsStringOrFunctionNode(groups) {
```

Comment:
Please move that to utils and rename to `isFunction` (maybe best name)

Reply:
I'll see what I can do, but note that these new utility functions, along with some existing ones, aren't very portable.
For example, I wanted to use isURLFunctionNode on line 27, instead of manually checking the type and value, but isURLFunctionNode only works in the printer, where the parser has already processed the nodes and converted "func" to "value-func".
Should I be specifying somewhere whether these will work in the parser, printer, or both (or should there be separate utils for these cases)?

Reply:
You can use name `isGroupListLikeStringOrFunction` (maybe less name)

Reply:
Moved to utils and renamed to `hasStringOrFunction`.